### PR TITLE
Fix reference counting error in enkf_fs

### LINKED ISSUE #737: Unstable tests
Several tests appear to influence each other, some, not all, of these are marked as `pytest.mark.unstable`.

### PR DESCRIPTION
**Pull request by mortalisk**
Resolves #737 

enkf_fs_get_ref in enkf_fs.cpp increaments the reference when it probably shouldn't. This caused the counter to be incremented when python created a reference to it, but python would not call the decrement when the python object was decremented, since it was supposedly a reference (that means some sort of weak reference in cwrap language).

The fix currently avoid this by making python use the strong reference functionality in stead, so that it will both increment and decrement the counter correctly.

### CODE REVIEW

File: python/res/enkf/enkf_fs_manager.py [hunk 1]
```diff
@@ -33,7 +33,7 @@ def addFileSystem(self, file_system, full_name):
     def dropOldestFileSystem(self):
         if len(self._fs_list) > 0:
             case_name = self._fs_list[0]
-            fs = self._fs_map[case_name]
+            #fs = self._fs_map[case_name]
```
Comment (sondreso) [inline python/res/enkf/enkf_fs_manager.py]: Delete or uncomment

**Comment by pinkwah** [comment]
This PR isn't complete: We should make sure that it isn't possible to use the "weak" EnkfFs in the future. Remove `weakref` from `EnkfFs` and throw an exception in `EnkFs.__init__` if `self.isReference() is True`

The bad `enkf_fs_ref` is used here also:
https://github.com/equinor/libres/blob/149b7f6aea1fa563c1832e86f8356101d1043b95/python/res/enkf/ert_run_context.py#L41-L42

Edit:  after investigation, it seems these two actually are weak references, so ignore this

File: python/tests/res/enkf/test_enkf_fs_manager1.py [hunk 2]
```diff
@@ -38,10 +38,7 @@ def test_create(self):
 
             fs2 = fsm.getFileSystem("newFS")
             self.assertEqual(2, fsm.getFileSystemCount())
-            self.assertEqual(1, fs2.refCount())
-
-            with self.assertRaises(AssertionError):
-                fs2.umount( )
+            #self.assertEqual(1, fs2.refCount())
```
Comment (pinkwah) [inline python/tests/res/enkf/test_enkf_fs_manager1.py]: Still have to do something about these asserts.

Comment (mortalisk) [reply python/tests/res/enkf/test_enkf_fs_manager1.py]: I deleted those. I think it is sufficiently tested in test_enkf_fs_manager2.py

**Review by markusdregi — APPROVED** [review]
You can merge this to the tunes of _Simply the best_ :musical_note:

**Review by jondequinor — APPROVED** [review]
LGTM

**Pull request merged**